crucible-llvm/src/Lang/Crucible/LLVM/ArraySizeProfile.hs

```diff
@@ -52,40 +53,41 @@ import qualified Lang.Crucible.LLVM.MemModel as C
 
 import qualified Lang.Crucible.LLVM.MemModel.Generic as G
 
-import What4.Interface (IsExpr, IsExprBuilder, SymExpr, asNat, asUnsignedBV)
+import qualified What4.Interface as W4
 
 type Profile = (Text, [[Maybe Int]])
 
 ptrStartsAlloc ::
-  IsExpr (SymExpr sym) =>
+  W4.IsExpr (W4.SymExpr sym) =>
   C.LLVMPtr sym w ->
   Maybe ()
-ptrStartsAlloc (C.llvmPointerView -> (_, asUnsignedBV -> Just 0)) = Just ()
+ptrStartsAlloc (C.llvmPointerView -> (_, W4.asUnsignedBV -> Just 0)) = Just ()
 ptrStartsAlloc _ = Nothing
 
 ptrAllocSize ::
-  forall sym w. IsExpr (SymExpr sym) =>
+  forall sym w. W4.IsExpr (W4.SymExpr sym) =>
   [G.MemAlloc sym] ->
   C.LLVMPtr sym w ->
   Maybe Int
 ptrAllocSize mem (C.llvmPointerView -> (blk, _)) = msum $ inAlloc <$> mem
   where inAlloc :: G.MemAlloc sym -> Maybe Int
         inAlloc (G.Alloc _ _ Nothing _ _ _) = Nothing
-        inAlloc (G.Alloc _ a (Just size) _ _ _) = do
-          blk' <- asNat blk
+        inAlloc (G.Alloc _ a (Just sz) _ _ _) = do
+          blk' <- W4.asNat blk
           if a == blk'
-            then fromIntegral <$> asUnsignedBV size
+            then fromIntegral <$> W4.asUnsignedBV sz
             else Nothing
         inAlloc _ = Nothing
 
-ptrIsArray :: IsExpr (SymExpr sym) =>
-              [G.MemAlloc sym] ->
-              C.LLVMPtr sym w ->
-              Maybe Int
+ptrIsArray ::
```

Comment:
I usually think of functions with names containing the word `Is` as predicates, i.e. returning booleans. The name doesn't make the significance of the return value clear.

Reply:
This same comment applies below. Since all of these functions are used only once by the functions after them, it might be more appropriate to put them in a `where` clause.

Reply:
Agreed entirely, these names have been irritating me for the same reason. It's kind of a weird situation where the functions are basically predicates, just with some extra data attached to true. Wanting to use the `Monad` instance for `Maybe` has led to some similarly-weird design choices, e.g. `ptrStartsAlloc` returning `Maybe ()` rather than `Bool`.

I think I'll change the names to be less predicate-like and just have `ptrStartsAlloc` return `Bool`, unless you have another idea?

Reply:
@chameco That makes sense to me!